src/all/nijiero/src/eu/kanade/tachiyomi/extension/all/nijiero/Nijiero.kt

```diff
@@ -0,0 +1,165 @@
+    override fun latestUpdatesParse(response: Response) = throw UnsupportedOperationException()
+
+    override fun popularMangaRequest(page: Int): Request {
+        val uniqueParam = System.currentTimeMillis()
+        val url = "$baseUrl/ranking.html?refresh=$uniqueParam"
+
+        return GET(url, headers)
+    }
+    override fun popularMangaParse(response: Response): MangasPage {
+        val document = response.asJsoup()
+        val mangas = document.selectFirst("#mainContent .allRunkingArea.tabContent.cf")!!.children().mapNotNull { it ->
+            val link = it.select("a")
+            if (!link.isEmpty()) {
+                SManga.create().apply {
+                    url = link.attr("href").removePrefix(baseUrl)
+                    title = link.attr("title")
+                    thumbnail_url = it.selectFirst("img")?.attr("src")
+                    status = SManga.COMPLETED
+                }
+            } else {
+                null
+            }
+        }
+
+        return MangasPage(mangas, false)
+    }
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        val tagIdx: Int = (filters.last() as TagFilter).state
+
+        val pageType = when {
+            query.isBlank() -> "category"
+            else -> "tag"
+        }
+
+        val keyword = when {
+            query.isBlank() -> nijieroTags[tagIdx]
+            else -> query
+        }.replace(Regex("""\s+"""), "-").lowercase()
+
+        val uniqueParam = System.currentTimeMillis()
+        val url = "$baseUrl/$pageType/$keyword/page/$page?refresh=$uniqueParam"
+
+        return GET(url, headers)
+    }
+    override fun searchMangaParse(response: Response): MangasPage {
+        val document = response.asJsoup()
+        var mangas = document.selectFirst(".contentList")!!.children().mapNotNull { it ->
+            val link = it.select("a")
+            if (!link.isEmpty()) {
+                SManga.create().apply {
+                    url = link.attr("href").removePrefix(baseUrl)
+                    title = link.attr("title")
+                    thumbnail_url = it.selectFirst("img")?.attr("data-src")
+                    status = SManga.COMPLETED
+                }
+            } else {
+                null
+            }
+        }
+        var isLastPage = document.selectFirst(Evaluator.Class("next page-numbers")) == null
+
+        if (mangas.isEmpty()) {
+            val currentUrl = response.request.url.toString()
+            val newUrl = if (currentUrl.contains("category/")) {
+                currentUrl.replace("category/", "tag/")
+            } else if (currentUrl.contains("tag/")) {
+                currentUrl.replace("tag/", "category/")
+            } else {
+                currentUrl
+            }
+            val newResponse = client.newCall(GET(newUrl, headers)).execute()
+            return searchMangaParse(newResponse)
```

Comment:
So if it doesn't find anything from either endpoint, it will recursively create new requests? You should have a guard against this, or change the implementation so that isn't possible.

Reply:
is it better?

Reply:
Other maintainers might have some additional suggestions in regard to the implementation details, but it seems fine - at least it's not recursively creating new requests.